src/Symfony/Component/HttpClient/MockHttpClient.php

```diff
@@ -61,6 +61,9 @@ public function request(string $method, string $url, array $options = []): Respo
             $response = new MockResponse();
         } elseif (\is_callable($this->responseFactory)) {
             $response = ($this->responseFactory)($method, $url, $options);
+            if (!$response instanceof ResponseInterface) {
```

Comment:
shouldn't this be moved outside of the parent if/elseif block?
the "else" below would need a similar check, isn't it?

Reply:
We can add a check for iterators too but I think making the error in this case is more unlikely. Do you want me to add it?

Reply:
Moving the check outside and below the exist if/elseif/else would work, isn't it?